{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/data_sata_2t/downloads/test\n"
     ]
    }
   ],
   "source": [
    "cd /mnt/data_sata_2t/downloads/test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "--2019-03-20 10:36:59--  http://people.virginia.edu/~wc9c/KING/Linux-king.tar.gz\n",
      "Resolving people.virginia.edu (people.virginia.edu)... 128.143.22.180\n",
      "Connecting to people.virginia.edu (people.virginia.edu)|128.143.22.180|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 799931 (781K) [application/x-gzip]\n",
      "Saving to: ‘Linux-king.tar.gz’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  6% 73.8K 10s\n",
      "    50K .......... .......... .......... .......... .......... 12%  111K 8s\n",
      "   100K .......... .......... .......... .......... .......... 19%  221K 6s\n",
      "   150K .......... .......... .......... .......... .......... 25%  112K 5s\n",
      "   200K .......... .......... .......... .......... .......... 32%  113K 5s\n",
      "   250K .......... .......... .......... .......... .......... 38%  222K 4s\n",
      "   300K .......... .......... .......... .......... .......... 44%  113K 4s\n",
      "   350K .......... .......... .......... .......... .......... 51%  114K 3s\n",
      "   400K .......... .......... .......... .......... .......... 57%  220K 3s\n",
      "   450K .......... .......... .......... .......... .......... 64%  113K 2s\n",
      "   500K .......... .......... .......... .......... .......... 70%  115K 2s\n",
      "   550K .......... .......... .......... .......... .......... 76%  212K 1s\n",
      "   600K .......... .......... .......... .......... .......... 83%  113K 1s\n",
      "   650K .......... .......... .......... .......... .......... 89%  219K 1s\n",
      "   700K .......... .......... .......... .......... .......... 96%  114K 0s\n",
      "   750K .......... .......... .......... .                    100% 65.5K=6.3s\n",
      "\n",
      "2019-03-20 10:37:06 (125 KB/s) - ‘Linux-king.tar.gz’ saved [799931/799931]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "wget -c http://people.virginia.edu/~wc9c/KING/Linux-king.tar.gz\n",
    "mv Linux-king.tar.gz ~/biosoftwares\n",
    "tar zxvf ~/biosoftwares/Linux-king.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.15\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf trios.vcf\n",
      "\t--out trios.vcftools\n",
      "\t--relatedness2\n",
      "\n",
      "After filtering, kept 3 out of 3 Individuals\n",
      "Outputting Individual Relatedness\n",
      "After filtering, kept 48642 out of a possible 48642 Sites\n",
      "Run Time = 0.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "vcftools --vcf trios.vcf --relatedness2 --out trios.vcftools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>INDV1</th>\n",
       "      <th>INDV2</th>\n",
       "      <th>N_AaAa</th>\n",
       "      <th>N_AAaa</th>\n",
       "      <th>N1_Aa</th>\n",
       "      <th>N2_Aa</th>\n",
       "      <th>RELATEDNESS_PHI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HY18119189</td>\n",
       "      <td>HY18119189</td>\n",
       "      <td>22851</td>\n",
       "      <td>0</td>\n",
       "      <td>22851</td>\n",
       "      <td>22851</td>\n",
       "      <td>0.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HY18119189</td>\n",
       "      <td>HY18119190</td>\n",
       "      <td>12615</td>\n",
       "      <td>177</td>\n",
       "      <td>22851</td>\n",
       "      <td>22645</td>\n",
       "      <td>0.269496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HY18119189</td>\n",
       "      <td>HY18119191</td>\n",
       "      <td>13057</td>\n",
       "      <td>181</td>\n",
       "      <td>22851</td>\n",
       "      <td>22969</td>\n",
       "      <td>0.277062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HY18119190</td>\n",
       "      <td>HY18119189</td>\n",
       "      <td>12615</td>\n",
       "      <td>177</td>\n",
       "      <td>22645</td>\n",
       "      <td>22851</td>\n",
       "      <td>0.269496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HY18119190</td>\n",
       "      <td>HY18119190</td>\n",
       "      <td>22645</td>\n",
       "      <td>0</td>\n",
       "      <td>22645</td>\n",
       "      <td>22645</td>\n",
       "      <td>0.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HY18119190</td>\n",
       "      <td>HY18119191</td>\n",
       "      <td>9819</td>\n",
       "      <td>3368</td>\n",
       "      <td>22645</td>\n",
       "      <td>22969</td>\n",
       "      <td>0.067589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HY18119191</td>\n",
       "      <td>HY18119189</td>\n",
       "      <td>13057</td>\n",
       "      <td>181</td>\n",
       "      <td>22969</td>\n",
       "      <td>22851</td>\n",
       "      <td>0.277062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HY18119191</td>\n",
       "      <td>HY18119190</td>\n",
       "      <td>9819</td>\n",
       "      <td>3368</td>\n",
       "      <td>22969</td>\n",
       "      <td>22645</td>\n",
       "      <td>0.067589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>HY18119191</td>\n",
       "      <td>HY18119191</td>\n",
       "      <td>22969</td>\n",
       "      <td>0</td>\n",
       "      <td>22969</td>\n",
       "      <td>22969</td>\n",
       "      <td>0.500000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        INDV1       INDV2  N_AaAa  N_AAaa  N1_Aa  N2_Aa  RELATEDNESS_PHI\n",
       "0  HY18119189  HY18119189   22851       0  22851  22851         0.500000\n",
       "1  HY18119189  HY18119190   12615     177  22851  22645         0.269496\n",
       "2  HY18119189  HY18119191   13057     181  22851  22969         0.277062\n",
       "3  HY18119190  HY18119189   12615     177  22645  22851         0.269496\n",
       "4  HY18119190  HY18119190   22645       0  22645  22645         0.500000\n",
       "5  HY18119190  HY18119191    9819    3368  22645  22969         0.067589\n",
       "6  HY18119191  HY18119189   13057     181  22969  22851         0.277062\n",
       "7  HY18119191  HY18119190    9819    3368  22969  22645         0.067589\n",
       "8  HY18119191  HY18119191   22969       0  22969  22969         0.500000"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_table('trios.vcftools.relatedness2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90p 64-bit (25 Mar 2016)          https://www.cog-genomics.org/plink2\n",
      "(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to trios.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out trios\n",
      "  --vcf trios.vcf\n",
      "\n",
      "32122 MB RAM detected; reserving 16061 MB for main workspace.\n",
      "--vcf: trios-temporary.bed + trios-temporary.bim + trios-temporary.fam written.\n",
      "48642 variants loaded from .bim file.\n",
      "3 people (0 males, 0 females, 3 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to trios.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 3 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997217.\n",
      "48642 variants and 3 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to trios.bed + trios.bim + trios.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "KING 2.1.8 - (c) 2010-2019 Wei-Min Chen\n",
      "\n",
      "The following parameters are in effect:\n",
      "                   Binary File :       trios.bed (-bname)\n",
      "\n",
      "Additional Options\n",
      "         Close Relative Inference : --related, --duplicate\n",
      "   Pairwise Relatedness Inference : --kinship [ON], --ibdseg, --ibs, --homog\n",
      "              Inference Parameter : --degree\n",
      "         Relationship Application : --unrelated, --cluster, --build\n",
      "                        QC Report : --bysample, --bySNP, --roh, --autoQC [ON]\n",
      "                     QC Parameter : --callrateN, --callrateM\n",
      "             Population Structure : --pca, --mds\n",
      "              Structure Parameter : --projection\n",
      "              Disease Association : --tdt\n",
      "   Quantitative Trait Association : --mtscore\n",
      "                Association Model : --trait [], --covariate []\n",
      "            Association Parameter : --invnorm, --maxP\n",
      "               Genetic Risk Score : --risk, --model [], --prevalence, --noflip\n",
      "              Computing Parameter : --cpus\n",
      "                   Optional Input : --fam [], --bim [], --sexchr [23]\n",
      "                           Output : --prefix [trios.king], --rplot\n",
      "\n",
      "The following analyses will run separately:  --kinship --autoQC\n",
      "\n",
      "KING starts at Wed Mar 20 11:28:33 2019\n",
      "Loading genotype data in PLINK binary format...\n",
      "Read in PLINK fam file trios.fam...\n",
      "  PLINK pedigrees loaded: 3 samples\n",
      "Read in PLINK bim file trios.bim...\n",
      "  Genotype data consist of 47665 autosome SNPs, 908 X-chromosome SNPs, 12 Y-chromosome SNPs, 57 mitochondrial SNPs\n",
      "  PLINK maps loaded: 48642 SNPs\n",
      "Read in PLINK bed file trios.bed...\n",
      "  PLINK binary genotypes loaded.\n",
      "  KING format genotype data successfully converted.\n",
      "Autosome genotypes stored in 2980 words for each of 3 individuals.\n",
      "\n",
      "Options in effect:\n",
      "\t--kinship\n",
      "\t--prefix trios.king\n",
      "\n",
      "Each family consists of one individual.\n",
      "Relationship inference across families starts at Wed Mar 20 11:28:33 2019\n",
      "8 CPU cores are used.\n",
      "                                         ends at Wed Mar 20 11:28:33 2019\n",
      "Between-family kinship data saved in file trios.king.kin0\n",
      "\n",
      "X-chromosome analysis...\n",
      "X-chromosome genotypes stored in 57 words for each of 3 individuals.\n",
      "Within-family kinship data saved in file trios.kingX.kin\n",
      "Relationship inference across families starts at Wed Mar 20 11:28:33 2019\n",
      "                                         ends at Wed Mar 20 11:28:33 2019\n",
      "Between-family kinship data saved in file trios.kingX.kin0\n",
      "Autosome genotypes stored in 2980 words for each of 3 individuals.\n",
      "\n",
      "Options in effect:\n",
      "\t--autoQC\n",
      "\t--prefix trios.king\n",
      "\n",
      "Auto-QC starts at Wed Mar 20 11:28:33 2019\n",
      "\n",
      "Auto-QC step 1: Apply SNP call rate filter 80.0% on 47665 SNPs (in 3 samples)\n",
      "  300 autosome SNPs have call rate < 80.0%\n",
      "  10 X-chr SNPs have call rate < 80.0%\n",
      "  8765 SNPs are monomorphic\n",
      "\n",
      "Auto-QC step 2: Apply sample call rate filter 95.0% on 3 samples (with 38868 SNPs)\n",
      "  0 samples have call rate < 95.0%\n",
      "\n",
      "Auto-QC step 3: Apply SNP call rate filter 95.0% on 38868 SNPs (in 3 samples)\n",
      "  0 SNPs have call rate < 95.0%\n",
      "  0 chr-X SNPs have call rate < 95.0%\n",
      "\n",
      "Auto-QC step 4: Apply call rate filters on 12 Y-chr SNPs\n",
      "\n",
      "  Step 4a: Apply Y-chr call rate filter 80.0% in males\n",
      "  0 chr-Y SNPs have call rate < 80.0% in males\n",
      "\n",
      "  Step 4b: Apply X-chr heterozygosity filter 5% in males\n",
      "  0 X-chr SNPs have heterozygosity > 5% in males\n",
      "\n",
      "  Step 4c: Apply Y-chr call rate filter 10% in females\n",
      "  0 chr-Y SNPs have call rate > 10% in females\n",
      "\n",
      "Auto-QC step 5: Gender QC on 3 samples\n",
      "\n",
      "  Step 5a: Determine thresholds in 10 Y-chr SNPs for gender checking\n",
      "  5.0 is used as the cutoff value for the chr-Y SNP count between males vs females\n",
      "\n",
      "  Step 5b: Gender checking\n",
      "  X-chr heterozygosity is set as 0.10\n",
      "  0 (reported) males have gender errors (less than 5.0 Y-chr SNPs)\n",
      "  0 (reported) females have gender errors (more than 5.0 Y-chr SNPs)\n",
      "  0 samples have additional gender errors (according to X-Chr heterozygosity filter 0.100)\n",
      "  Genders of 3 samples are not reported but will be inferred now\n",
      "  Gender plot are generated in trios.king_gender_autoplot.pdf\n",
      "\n",
      "Auto-QC step 6: Apply call rate filters on 10 Y-chr SNPs\n",
      "\n",
      "  Step 6a: Apply Y-chr call rate filter 95.0% in males\n",
      "  0 chr-Y SNPs have call rate < 95.0% in males\n",
      "\n",
      "  Step 6b: Apply X-chr heterozygosity filter 1% in males\n",
      "  0 X-chr SNPs have heterozygosity > 1% in males\n",
      "\n",
      "  Step 6c: Apply Y-chr call rate filter 2% in females\n",
      "  0 chr-Y SNPs have call rate > 2% in females\n",
      "\n",
      "Auto-QC step 7: Final check\n",
      "  3 samples, 38868 autosome SNPs, 632 X-chr SNPs, 10 Y-chr SNPs\n",
      "\n",
      "Auto-QC step 8: QC Summary Report\n",
      "\n",
      "Step Description                                            Subjects  SNPs      \n",
      "1    Raw data counts                                        3         48642     \n",
      "1.1  SNPs with very low call rate < 80% (removed)                     (310)\n",
      "1.2  Monomorphic SNPs (removed)                                       (8765)\n",
      "1.3  Sample call rate < 95% (removed)                       (0)\n",
      "1.4  SNPs with call rate < 95% (removed)                              (0)\n",
      "2    data counts for gender error checking                  3         39567     \n",
      "2.1  Y-chr SNPs with call rate < 80% in men (removed)                 (0)\n",
      "2.2  X-chr SNPs with heterozygosity > 5% in men (removed)             (0)\n",
      "2.3  Y-chr SNPs with genotypes in >10% women (removed)                (0)\n",
      "2.4  Mislabeled as male (removed)                           (0)\n",
      "2.5  Mislabeled as female (removed)                         (0)\n",
      "2.6  Suspicious gender error (removed)                      (0)\n",
      "2.7  Y-chr SNPs with call rate < 95% in men (removed)                 (0)\n",
      "2.8  X-chr SNPs with heterozygosity > 1% in men (removed)             (0)\n",
      "2.9  Y-chr SNPs with genotypes in >2% women (removed)                 (0)\n",
      "3    Generate Final Study Files                             \n",
      "     Final QC'ed data                                       3         39567     \n",
      "\n",
      "QC summary report saved in trios.king_autoQC_Summary.txt\n",
      "SNP-removal QC file saved in trios.king_autoQC_snptoberemoved.txt\n",
      "Sample-removal QC file saved in trios.king_autoQC_sampletoberemoved.txt\n",
      "Update-sex QC file saved in trios.king_autoQC_updatesex.txt\n",
      "\n",
      "Auto-QC ends at Wed Mar 20 11:28:34 2019\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "plink2 --vcf trios.vcf --make-bed --out trios\n",
    "~/biosoftwares/king -b trios.bed --kinship --autoQC --prefix trios.king"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID1</th>\n",
       "      <th>ID1</th>\n",
       "      <th>FID2</th>\n",
       "      <th>ID2</th>\n",
       "      <th>N_SNP</th>\n",
       "      <th>HetHet</th>\n",
       "      <th>IBS0</th>\n",
       "      <th>Kinship</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HY18119189</td>\n",
       "      <td>HY18119189</td>\n",
       "      <td>HY18119190</td>\n",
       "      <td>HY18119190</td>\n",
       "      <td>47444</td>\n",
       "      <td>0.2637</td>\n",
       "      <td>0.0029</td>\n",
       "      <td>0.2727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HY18119189</td>\n",
       "      <td>HY18119189</td>\n",
       "      <td>HY18119191</td>\n",
       "      <td>HY18119191</td>\n",
       "      <td>47439</td>\n",
       "      <td>0.2704</td>\n",
       "      <td>0.0037</td>\n",
       "      <td>0.2772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HY18119190</td>\n",
       "      <td>HY18119190</td>\n",
       "      <td>HY18119191</td>\n",
       "      <td>HY18119191</td>\n",
       "      <td>47435</td>\n",
       "      <td>0.2048</td>\n",
       "      <td>0.0674</td>\n",
       "      <td>0.0731</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         FID1         ID1        FID2         ID2  N_SNP  HetHet    IBS0  \\\n",
       "0  HY18119189  HY18119189  HY18119190  HY18119190  47444  0.2637  0.0029   \n",
       "1  HY18119189  HY18119189  HY18119191  HY18119191  47439  0.2704  0.0037   \n",
       "2  HY18119190  HY18119190  HY18119191  HY18119191  47435  0.2048  0.0674   \n",
       "\n",
       "   Kinship  \n",
       "0   0.2727  \n",
       "1   0.2772  \n",
       "2   0.0731  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_table('trios.king.kin0')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KING 2.1.8 - (c) 2010-2019 Wei-Min Chen\n",
      "\n",
      "The following parameters are in effect:\n",
      "                   Binary File :       trios.bed (-bname)\n",
      "\n",
      "Additional Options\n",
      "         Close Relative Inference : --related, --duplicate\n",
      "   Pairwise Relatedness Inference : --kinship, --ibdseg [ON], --ibs, --homog\n",
      "              Inference Parameter : --degree\n",
      "         Relationship Application : --unrelated, --cluster, --build\n",
      "                        QC Report : --bysample, --bySNP, --roh, --autoQC\n",
      "                     QC Parameter : --callrateN, --callrateM\n",
      "             Population Structure : --pca, --mds\n",
      "              Structure Parameter : --projection\n",
      "              Disease Association : --tdt\n",
      "   Quantitative Trait Association : --mtscore\n",
      "                Association Model : --trait [], --covariate []\n",
      "            Association Parameter : --invnorm, --maxP\n",
      "               Genetic Risk Score : --risk, --model [], --prevalence, --noflip\n",
      "              Computing Parameter : --cpus\n",
      "                   Optional Input : --fam [], --bim [], --sexchr [23]\n",
      "                           Output : --prefix [king], --rplot\n",
      "\n",
      "KING starts at Wed Mar 20 11:18:40 2019\n",
      "Loading genotype data in PLINK binary format...\n",
      "Read in PLINK fam file trios.fam...\n",
      "  PLINK pedigrees loaded: 3 samples\n",
      "Read in PLINK bim file trios.bim...\n",
      "  Genotype data consist of 47665 autosome SNPs, 908 X-chromosome SNPs, 12 Y-chromosome SNPs, 57 mitochondrial SNPs\n",
      "  PLINK maps loaded: 48642 SNPs\n",
      "Read in PLINK bed file trios.bed...\n",
      "  PLINK binary genotypes loaded.\n",
      "  KING format genotype data successfully converted.\n",
      "\n",
      "--kinship analysis carried out instead for such a small sample size.\n",
      "Autosome genotypes stored in 745 words for each of 3 individuals.\n",
      "\n",
      "Options in effect:\n",
      "\t--kinship\n",
      "\n",
      "Each family consists of one individual.\n",
      "Relationship inference across families starts at Wed Mar 20 11:18:40 2019\n",
      "8 CPU cores are used.\n",
      "                                         ends at Wed Mar 20 11:18:40 2019\n",
      "Between-family kinship data saved in file king.kin0\n",
      "Note --kinship --degree <n> can filter & speed up the kinship computing.\n",
      "\n",
      "X-chromosome analysis...\n",
      "X-chromosome genotypes stored in 0 words for each of 3 individuals.\n",
      "Within-family kinship data saved in file kingX.kin\n",
      "Relationship inference across families starts at Wed Mar 20 11:18:40 2019\n",
      "                                         ends at Wed Mar 20 11:18:40 2019\n",
      "Between-family kinship data saved in file kingX.kin0\n",
      "KING 2.1.8 - (c) 2010-2019 Wei-Min Chen\n",
      "\n",
      "The following parameters are in effect:\n",
      "                   Binary File :       trios.bed (-bname)\n",
      "\n",
      "Additional Options\n",
      "         Close Relative Inference : --related, --duplicate\n",
      "   Pairwise Relatedness Inference : --kinship, --ibdseg [ON], --ibs, --homog\n",
      "              Inference Parameter : --degree\n",
      "         Relationship Application : --unrelated, --cluster, --build\n",
      "                        QC Report : --bysample, --bySNP, --roh, --autoQC\n",
      "                     QC Parameter : --callrateN, --callrateM\n",
      "             Population Structure : --pca, --mds\n",
      "              Structure Parameter : --projection\n",
      "              Disease Association : --tdt\n",
      "   Quantitative Trait Association : --mtscore\n",
      "                Association Model : --trait [], --covariate []\n",
      "            Association Parameter : --invnorm, --maxP\n",
      "               Genetic Risk Score : --risk, --model [], --prevalence, --noflip\n",
      "              Computing Parameter : --cpus\n",
      "                   Optional Input : --fam [], --bim [], --sexchr [23]\n",
      "                           Output : --prefix [trios.king], --rplot [ON]\n",
      "\n",
      "KING starts at Wed Mar 20 11:18:40 2019\n",
      "Loading genotype data in PLINK binary format...\n",
      "Read in PLINK fam file trios.fam...\n",
      "  PLINK pedigrees loaded: 3 samples\n",
      "Read in PLINK bim file trios.bim...\n",
      "  Genotype data consist of 47665 autosome SNPs, 908 X-chromosome SNPs, 12 Y-chromosome SNPs, 57 mitochondrial SNPs\n",
      "  PLINK maps loaded: 48642 SNPs\n",
      "Read in PLINK bed file trios.bed...\n",
      "  PLINK binary genotypes loaded.\n",
      "  KING format genotype data successfully converted.\n",
      "\n",
      "--kinship analysis carried out instead for such a small sample size.\n",
      "Autosome genotypes stored in 745 words for each of 3 individuals.\n",
      "\n",
      "Options in effect:\n",
      "\t--kinship\n",
      "\t--prefix trios.king\n",
      "\n",
      "Each family consists of one individual.\n",
      "Relationship inference across families starts at Wed Mar 20 11:18:40 2019\n",
      "8 CPU cores are used.\n",
      "                                         ends at Wed Mar 20 11:18:40 2019\n",
      "Between-family kinship data saved in file trios.king.kin0\n",
      "Note --kinship --degree <n> can filter & speed up the kinship computing.\n",
      "\n",
      "X-chromosome analysis...\n",
      "X-chromosome genotypes stored in 0 words for each of 3 individuals.\n",
      "Within-family kinship data saved in file trios.kingX.kin\n",
      "Relationship inference across families starts at Wed Mar 20 11:18:40 2019\n",
      "                                         ends at Wed Mar 20 11:18:40 2019\n",
      "Between-family kinship data saved in file trios.kingX.kin0\n",
      "Please use --degree, --degree 2, or --degree 3 to reduce #relatives.\n",
      "  IBD1 vs IBD2 plot is generated in trios.king_ibd1vsibd2.pdf\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "~/biosoftwares/king -b trios.bed --ibdseg\n",
    "~/biosoftwares/king -b trios.bed --ibdseg --rplot --prefix trios.king"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "        <iframe\n",
       "            width=\"600\"\n",
       "            height=\"300\"\n",
       "            src=\"./trios.king_ibd1vsibd2.pdf\"\n",
       "            frameborder=\"0\"\n",
       "            allowfullscreen\n",
       "        ></iframe>\n",
       "        "
      ],
      "text/plain": [
       "<IPython.lib.display.IFrame at 0x7f4ced5da4e0>"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from IPython.display import IFrame\n",
    "IFrame(\"./trios.king_ibd1vsibd2.pdf\", width=600, height=300)\n",
    "#IFrame(\"./trios.king_gender_autoplot.pdf\", width=600, height=300)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KING 2.1.8 - (c) 2010-2019 Wei-Min Chen\n",
      "\n",
      "The following parameters are in effect:\n",
      "                   Binary File :       trios.bed (-bname)\n",
      "\n",
      "Additional Options\n",
      "         Close Relative Inference : --related [ON], --duplicate\n",
      "   Pairwise Relatedness Inference : --kinship, --ibdseg, --ibs, --homog\n",
      "              Inference Parameter : --degree\n",
      "         Relationship Application : --unrelated, --cluster, --build\n",
      "                        QC Report : --bysample, --bySNP, --roh, --autoQC\n",
      "                     QC Parameter : --callrateN, --callrateM\n",
      "             Population Structure : --pca, --mds\n",
      "              Structure Parameter : --projection\n",
      "              Disease Association : --tdt\n",
      "   Quantitative Trait Association : --mtscore\n",
      "                Association Model : --trait [], --covariate []\n",
      "            Association Parameter : --invnorm, --maxP\n",
      "               Genetic Risk Score : --risk, --model [], --prevalence, --noflip\n",
      "              Computing Parameter : --cpus\n",
      "                   Optional Input : --fam [trios.fam], --bim [trios.bim],\n",
      "                                    --sexchr [23]\n",
      "                           Output : --prefix [trios.king], --rplot\n",
      "\n",
      "KING starts at Wed Mar 20 11:23:18 2019\n",
      "Loading genotype data in PLINK binary format...\n",
      "Read in PLINK fam file trios.fam...\n",
      "  PLINK pedigrees loaded: 3 samples\n",
      "Read in PLINK bim file trios.bim...\n",
      "  Genotype data consist of 47665 autosome SNPs, 908 X-chromosome SNPs, 12 Y-chromosome SNPs, 57 mitochondrial SNPs\n",
      "  PLINK maps loaded: 48642 SNPs\n",
      "Read in PLINK bed file trios.bed...\n",
      "  PLINK binary genotypes loaded.\n",
      "  KING format genotype data successfully converted.\n",
      "\n",
      "--related is replaced with --kinship for a small sample size.\n",
      "Autosome genotypes stored in 745 words for each of 3 individuals.\n",
      "\n",
      "Options in effect:\n",
      "\t--kinship\n",
      "\t--prefix trios.king\n",
      "\n",
      "Each family consists of one individual.\n",
      "Relationship inference across families starts at Wed Mar 20 11:23:18 2019\n",
      "8 CPU cores are used.\n",
      "                                         ends at Wed Mar 20 11:23:18 2019\n",
      "Between-family kinship data saved in file trios.king.kin0\n",
      "Note --kinship --degree <n> can filter & speed up the kinship computing.\n",
      "\n",
      "X-chromosome analysis...\n",
      "X-chromosome genotypes stored in 0 words for each of 3 individuals.\n",
      "Within-family kinship data saved in file trios.kingX.kin\n",
      "Relationship inference across families starts at Wed Mar 20 11:23:18 2019\n",
      "                                         ends at Wed Mar 20 11:23:18 2019\n",
      "Between-family kinship data saved in file trios.kingX.kin0\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "~/biosoftwares/king -b trios.bed --fam trios.fam --bim trios.bim --related --prefix trios.king"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
